tests/code/test_minor.py

```diff
@@ -132,26 +134,345 @@ def test_getCommunityEngagementByTerm():
+            if key == "user":
+                actualValue = 'ramsayb2'
+            elif key == "term":
+                actualValue = 3
+            else:
+                actualValue = getattr(savedRequest, key)
+            assert actualValue == expectedValue
+
+        transaction.rollback()
+
+@pytest.mark.integration
+def test_setCommunityEngagementForUser():
+    with mainDB.atomic() as transaction: 
+        IndividualRequirement.delete().execute()
+
+        # Adding requirement
+        # TODO add existing
+        # TODO add nonexisting
+        #setCommunityEngagementForUser('add', engagementData, 'ramsayb2')
+        khattsEngagementData1 = {"id": 2,
+                                "matched": False, 
+                                "name": 'Spanish Help',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'khatts'}
+        
+        khattsEngagementData2 = {"id": 9,
+                                "matched": False, 
+                                "name": 'CELTS-Sponsored Event',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData3 = {"id": 6,
+                                "matched": False, 
+                                "name": 'Habitat For Humanity',
+                                'term': 2,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData4 = {"id": 4,
+                                "matched": False, 
+                                "name": 'People Who Care',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'khatts'}
+        
+        khattsEngagementData5 = {"id": 1,
+                                "matched": False, 
+                                "name": 'Databses',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'khatts'}
+        
+        neillzEngagementData1 = {"id": 4,
+                                "matched": False, 
+                                "name": 'People Who Care',
+                                'term': 3,
+                                "type": 'program',
+                                'username': 'neillz'}
+        
+        neillzEngagementData2 = {"id": 1,
+                                "matched": False, 
+                                "name": 'Databses',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'neillz'}
+        
+        setCommunityEngagementForUser('add', khattsEngagementData1, 'ramsayb2')
+        
+        khattsEngagements = IndividualRequirement.select()
```

Comment:
Why in this test are we selecting all IndividualRequirements and then checking the index. Couldn't we just use a where clause to make sure we are only getting the ones for the user we want?